Document relative link resolution in MDX files (#5384)
mintlify/docs: guides/linking.mdx

- [Custom components](/customize/react-components)
```

Mintlify resolves relative paths (`./` and `../`) based on the source file's location in your project directory. This works for links, images, and JSX elements like `<Card>` and `<a>` tags.

```mdx
- [Sibling page](./sibling-page)
- [Parent section page](../other-page)
```

For `index.mdx` files, relative paths resolve from the directory that contains the index file. For example, a `./setup` link in `guides/getting-started/index.mdx` resolves to `/guides/getting-started/setup`.

Links preserve fragments and query strings.

```mdx
[Setup instructions](./setup#step-1)
```

<Tip>

>   Root-relative paths (starting with `/`) are generally preferred for internal links because they remain correct when you move the linking page to a different directory. Use relative paths when you want links to stay valid if an entire directory subtree is moved together.

Reviewer: ```suggestion
  Root-relative paths (starting with `/`) perform better for internal links because they remain correct if you move the linking page to a different directory.
```

- [Custom components](/customize/react-components)
```

### Relative paths

> 

Reviewer: ```suggestion
```

Mintlify resolves relative paths (`./` and `../`) based on the source file's location in your project directory. This works for links, images, and JSX elements like `<Card>` and `<a>` tags.

```mdx
- [Sibling page](./sibling-page)
- [Parent section page](../other-page)
```

For `index.mdx` files, relative paths resolve from the directory that contains the index file. For example, a `./setup` link in `guides/getting-started/index.mdx` resolves to `/guides/getting-started/setup`.

> Fragments and query strings are preserved:

Reviewer: ```suggestion
Links preserve fragments and query strings.
```

<AccordionGroup>
<Accordion title="Should I use root-relative or absolute URLs for internal links?">
Root-relative paths (starting with `/`) are the most common choice for internal links in Mintlify. They work consistently regardless of where the linking page sits in your directory, and they don't break if your documentation domain changes. Absolute URLs for internal links create unnecessary brittleness.

> Relative paths (`./` and `../`) are also supported and resolve correctly based on the source file's location. They're useful when you want links within a directory subtree to stay valid if the subtree is moved as a unit.

Reviewer: ```suggestion
You can use relative paths (`./` and `../`), but because they resolve based on the source file's location they may break more frequently.
```
